diff --git a/index.d.mts b/index.d.mts
new file mode 100644
index 0000000000000000000000000000000000000000..ab633192d605ff3e6fcc2e502bf5a6e3ebd0e902
--- /dev/null
+++ b/index.d.mts
@@ -0,0 +1,22 @@
+// ./index.d.ts
+
+/**
+ * convert `str` to ms
+ */
+export type Units =
+  'nanosecond' | 'ns' |
+  'µs' | 'μs' | 'us' | 'microsecond' |
+  'millisecond' | 'ms' |
+  'second' | 'sec' | 's' |
+  'minute' | 'min' | 'm' |
+  'hour' | 'hr' | 'h' |
+  'day' | 'd' |
+  'week' | 'wk' | 'w' |
+  'month' | 'b' |
+  'year' | 'yr' | 'y'
+
+declare const parse: ((input: string, format?: Units) => number | undefined) & {
+  [key: string]: number;
+};
+
+export default parse;
diff --git a/index.d.ts b/index.d.ts
index 9164554b620c90b7d982152e127db727b3bcfa9c..ab633192d605ff3e6fcc2e502bf5a6e3ebd0e902 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -15,7 +15,7 @@ export type Units =
   'month' | 'b' |
   'year' | 'yr' | 'y'
 
-declare const parse: ((input: string, format?: Units) => number) & {
+declare const parse: ((input: string, format?: Units) => number | undefined) & {
   [key: string]: number;
 };
 
diff --git a/package.json b/package.json
index bbb3890cc04941b45518e421f20d76ecab1ca9cc..075336a3d7340afe0dc27136eb09db7bf64fe2d3 100644
--- a/package.json
+++ b/package.json
@@ -12,8 +12,7 @@
   "types": "./index.d.ts",
   "exports": {
     "require": "./index.js",
-    "import": "./index.mjs",
-    "types": "./index.d.ts"
+    "import": "./index.mjs"
   },
   "dependencies": {},
   "devDependencies": {